### PR TITLE
fix(race): fixing race condition with closer use + assign uids + bgindex test fix

### PR DESCRIPTION
**Pull request by aman-bansal**
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7266)

### CODE REVIEW

**Review by jarifibrahim — CHANGES_REQUESTED** [review]
Reduce the number of changes.

File: dgraph/cmd/zero/assign.go [hunk 1]
```diff
@@ -196,7 +196,7 @@ func (s *Server) AssignUids(ctx context.Context, num *pb.Num) (*pb.AssignedIds,
 
 	select {
 	case <-ctx.Done():
-		return reply, ctx.Err()
+		return &emptyAssignedIds, ctx.Err()
```
Comment (jarifibrahim) [inline dgraph/cmd/zero/assign.go:199]: why do you need to change this?

File: graphql/admin/shutdown.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ import (
 var (
 	// ServerCloser is used to signal and wait for other goroutines to return gracefully after user
 	// requests shutdown.
-	ServerCloser *z.Closer
+	ServerCloser = z.NewCloser(0)
```
Comment (jarifibrahim) [inline graphql/admin/shutdown.go:31]:
```suggestion
	ServerCloser z.NewCloser
```

File: systest/bgindex/docker-compose.yml [hunk 3]
```diff
@@ -19,109 +19,8 @@ services:
       source: ../../ee/acl/hmac-secret
       target: /secret/hmac
       read_only: true
-    command: /gobin/dgraph alpha --my=alpha1:7080 --zero=zero1:5080,zero2:5080,zero3:5080
```
Comment (jarifibrahim) [inline systest/bgindex/docker-compose.yml:22]: So this PR has 3 changes?
1. closer fix
2. Assign UID fix
3. bgindex fix?

Comment (aman-bansal) [reply systest/bgindex/docker-compose.yml:22]: yes, this PR has above three fix. let me change the description also

File: dgraph/cmd/zero/assign.go (hunk 1 above)
Comment (aman-bansal) [reply dgraph/cmd/zero/assign.go:199]: There is a race condition with the use of reply. This happens when ctx.Done was called but lease is still working to assign uids

**Review by jarifibrahim — APPROVED** [review]

**Pull request merged**